examples/ansible/setup_neuvector.yml

```diff
@@ -0,0 +1,113 @@
+# When 'neuvector_state' is set to 'absent', the playbook will:
+#   - Stop the NeuVector service
+#   - Execute nevector runlabel UNINSTALL
+#   - Remove the NeuVector image
+
+- name: Running the NeuVector workload
+  hosts: alphost
+  vars:
+    neuvector_state: present
+    workload:
+      name: neuvector
+      image: registry.opensuse.org/suse/alp/workloads/bci_containerfiles/suse/alp/workloads/neuvector-demo:latest
+
+  tasks:
+    - name: Setup NeuVector
+      when:
+        - neuvector_state == "present"
+      block:
+        - name: Check if workload {{ workload.name }} is already installed
+          become: true
+          containers.podman.podman_container_info:
+            name: "{{ workload.name }}-allinone"
+          register: workload_info
+          changed_when: false
+
+        - name: Set SELinux into permissive mode
+          become: true
+          ansible.posix.selinux:
+            policy: targeted
+            state: permissive
+          when:
+            - workload_info.containers | length == 0
+
+        - name: Retrieve image for workload {{ workload.name }}
+          become: true
+          containers.podman.podman_image:
+            name: "{{ workload.image }}"
+            state: present
+          register: retrieve_workload_image
+          when:
+            - workload_info.containers | length == 0
+
+        - name: Execute nevector runlabel INSTALL
+          become: true
+          ansible.builtin.command: >-
+            podman container runlabel install {{ workload.image }}
+          register: workload_runlabel_install
+          changed_when:
+            - ('already exist' not in workload_runlabel_install.stdout)
+            - workload_info.containers | length == 0
+
+        - name: Start NeuVector service
+          become: true
+          ansible.builtin.systemd:
+            name: neuvector.service
+            state: started
+          register: workload_service_start
+
+        - name: enable service to start on a reboot
+          become: true
+          ansible.builtin.systemd:
+            name: neuvector.service
+            enabled: yes
+          when:
+            - workload_service_start is success
+
+        - name: Print message connect to NeuVector
+          ansible.builtin.debug:
+            msg: >-
+              NeuVector is running on https://{{ ansible_default_ipv4.address }}:8443
+              You need to accept the warning about the self-signed SSL certificate
+              and log in with the following default credentials: admin / admin.
+          when:
+            - workload_service_start is success
+
+    - name: Remove NeuVector
+      when:
+        - neuvector_state == "absent"
+      block:
+        - name: Stop NeuVector service
+          become: true
+          ansible.builtin.systemd:
+            name: neuvector.service
+            state: stopped
+            enabled: no
+          ignore_errors: yes
```

Comment:
What is this for please?

Reply:
When 'neuvector_state' is set to 'absent', the playbook will, Stop the NeuVector service, Execute nevector runlabel UNINSTALL, Remove the NeuVector image

Reply:
But why do you ignore errors?

Reply:
To ensure subsequent tasks in the "Remove NeuVector" block are executed for other cleanup steps

Reply:
would you suggest using block, always, and handlers instead of ignore errors?

Reply:
If there is error the playbook should fail so the user can go and fix the error.
In ideal world there should be no error but if it occurs you should not ignore it unless you have specific reason for it.

Reply:
The error occurs when remove is run twice and it is unable to find the requested service in situations considering only the service is removed due to broken state etc, it can continue running label uninstall and image remove ie-

ansible-playbook setup_neuvector.yml -e neuvector_state=absent

TASK [Stop NeuVector service] *******************************************************************
fatal: [alphost]: FAILED! => {"changed": false, "msg": "Could not find the requested service neuvector.service: host"}
...ignoring

TASK [Execute nevector runlabel UNINSTALL] ******************************************************
changed: [alphost]

TASK [Remove image for workload neuvector] ******************************************************
changed: [alphost]

Reply:
^Got it will make the required changes